### PR TITLE
DLUHC-175 Handle text and number inputs

### PR DESCRIPTION
**Pull request by bethpritchard**
Adds text and number input to form 

image

image

### CODE REVIEW

File: dluhc-component-library/src/components/siteSelectionForm/components/NumberInput.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+interface NumberInputProps {
+  value: number;
+  onChange: (value: number) => void;
+}
+
+const NumberInput = ({ value, onChange }: NumberInputProps) => {
```
Comment (JohnEz) [inline dluhc-component-library/src/components/siteSelectionForm/components/NumberInput.tsx:6]: Is there a reason we cant have one "Input" component that passes through any props we need? Is there a benefit to having them as two components?

File: dluhc-component-library/src/components/siteSelectionForm/components/NumberInput.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+const NumberInput = ({ value, onChange }: NumberInputProps) => {
+  const numberInputBox = (
+    <div className="flex items-center mb-4">
+      <label className="font-semibold flex">
+        <input
+          type="number"
+          class="text mr-2"
+          step="1"
+          min="18"
```
Comment (JohnEz) [inline dluhc-component-library/src/components/siteSelectionForm/components/NumberInput.tsx:14]: step, min and max should be passed in, not hardcoded

File: dluhc-component-library/src/components/siteSelectionForm/components/DynamicForm.tsx [hunk 3]
```diff
@@ -11,14 +13,17 @@ interface DynamicFormProps {
...
+  NumberInput,
   MultiSelect,
   None,
 }
 
 const getInputType = (formPageSchema: FormPageSchema) => {
   if (formPageSchema.type === "string") {
-    return formPageSchema.enum ? InputType.MultiSelect : InputType.Input;
+    return formPageSchema.enum ? InputType.MultiSelect : InputType.TextInput;
+  } else if (formPageSchema.type === "number") {
```
Comment (JohnEz) [inline dluhc-component-library/src/components/siteSelectionForm/components/DynamicForm.tsx:25]: We dont need the "else" here, just do...

```
if (formPageSchema.type === "number") {
  return InputType.NumberInput;
}
```
below the first if.

The returns mean we dont fall through to the next one and its easier to read.

File: dluhc-component-library/src/stories/SiteSelection.stories.tsx [hunk 4]
```diff
@@ -52,6 +52,17 @@ export const Default = {
           type: "string",
           title: "Your name",
         },
+        age: {
+          type: "number",
+          title: "Age in years",
+          step: "1",
+          min: "18",
```
Comment (amjones-scottlogic) [inline dluhc-component-library/src/stories/SiteSelection.stories.tsx:59]: Looking at the JsonSchema docs these should be numbers not strings.

Also annoyingly there are two types. "number" and "integer". Lets ignore this for now though

**Review by samornsby — APPROVED** [review]

**Pull request merged**